crates/noirc_frontend/src/parser/parser.rs

```diff
@@ -841,6 +842,20 @@ fn string_type() -> impl NoirParser<UnresolvedType> {
         .map(UnresolvedType::String)
 }
 
+fn format_string_type(
+    type_parser: impl NoirParser<UnresolvedType>,
+) -> impl NoirParser<UnresolvedType> {
+    keyword(Keyword::FormatString)
+        .ignore_then(
+            type_expression()
+                .or_not()
```

Comment:
```suggestion
```
Why is the `N` in `fmtstr<N, T>` optional? It should be required

Reply:
Yeah it should not be optional. This is leftover when I thought I didn't want to require the user to specify the size, but that causes more headaches than it solves.

Reply:
I removed this without committing the suggestion